server/server.js

```diff
@@ -11,6 +13,101 @@ const db = new pg.Pool({
 });
 
 const app = express();
+app.use(express.json());
+app.use(errorMiddleware);
+
+const queryObject = {
+  get: `
+  select *
+    from "entries"
+    order by "entryId";
+  `,
+
+  post: `
+  insert into "entries" ("title", "notes", "photoUrl")
+    values ($1, $2, $3)
+  returning *;
+  `,
+
+  put: `
+  update "entries"
+  set "title"=$1,
+      "notes"=$2,
+      "photoUrl"=$3
+  where "entryId"=$4
+  returning *;
+  `,
+
+  delete: `
+  delete
+  from "entries"
+  where "entryId"=$1
+  returning *;
+  `,
+};
+
+// Routing functions
+app.get('/api/entries', async (req, res, next) => {
+  try {
+    const sql = queryObject.get;
+    const result = await db.query(sql);
+    res.status(200).json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/entries', async (req, res, next) => {
+  try {
+    const sql = queryObject.post;
+    const { title, notes, photoUrl } = req.body;
```

Comment:
It would be good in this endpoint (and the others that have a body) to validate that the body was sent correctly.

In theory, you know that your client will always call it correctly. However, it is possible for people to hijack your client and change how it works. So properly validating all the inputs is important to keep someone from corrupting your data.